registry/repository.go

```diff
@@ -143,5 +189,60 @@ func (c *repositoryClient) uploadSignatureManifest(ctx context.Context, subject,
 		PackImageManifest:   c.OCIImageManifest,
 	}
 
-	return oras.Pack(ctx, c.Repository, ArtifactTypeNotation, []ocispec.Descriptor{blobDesc}, opts)
+	return oras.Pack(ctx, c.GraphTarget, ArtifactTypeNotation, []ocispec.Descriptor{blobDesc}, opts)
+}
+
+// signatureReferrers returns referrer nodes of desc in target filtered by
+// the "application/vnd.cncf.notary.signature" artifact type
+func signatureReferrers(ctx context.Context, target content.ReadOnlyGraphStorage, desc ocispec.Descriptor) ([]ocispec.Descriptor, error) {
```

Comment:
can you please help me understand why do we need this change?

Reply:
Resolving as we had an offline discussion on it.